packages/block-editor/src/components/link-control/index.js

```diff
@@ -251,59 +369,121 @@ function LinkControl( {
 			}
 		);
 
-		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
-		const searchResultsLabelId = isInitialSuggestions
-			? `block-editor-link-control-search-results-label-${ instanceId }`
-			: undefined;
-		const labelText = isInitialSuggestions
-			? __( 'Recently updated' )
-			: sprintf( __( 'Search results for %s' ), inputValue );
+		const directLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const isSingleDirectEntryResult =
+			suggestions.length === 1 &&
+			directLinkEntryTypes.includes(
+				suggestions[ 0 ].type.toLowerCase()
+			);
+		const shouldShowCreateEntity =
+			createSuggestion &&
+			! isSingleDirectEntryResult &&
+			! isInitialSuggestions;
+
 		// According to guidelines aria-label should be added if the label
 		// itself is not visible.
 		// See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
-		const ariaLabel = isInitialSuggestions ? undefined : labelText;
-		const SearchResultsLabel = (
+		const searchResultsLabelId = `block-editor-link-control-search-results-label-${ instanceId }`;
+		const labelText = isInitialSuggestions
+			? __( 'Recently updated' )
+			: sprintf( __( 'Search results for "%s"' ), inputValue );
+
+		const searchResultsLabel = (
 			<span
-				className="block-editor-link-control__search-results-label"
+				className={ classnames(
+					'block-editor-link-control__search-results-label',
+					{
+						'screen-reader-text': ! isInitialSuggestions,
+					}
+				) }
 				id={ searchResultsLabelId }
-				aria-label={ ariaLabel }
 			>
 				{ labelText }
 			</span>
 		);
 
 		return (
 			<div className="block-editor-link-control__search-results-wrapper">
-				{ isInitialSuggestions ? (
-					SearchResultsLabel
-				) : (
-					<VisuallyHidden>{ SearchResultsLabel }</VisuallyHidden>
-				) }
-
+				{ searchResultsLabel }
 				<div
 					{ ...suggestionsListProps }
 					className={ resultsListClasses }
 					aria-labelledby={ searchResultsLabelId }
 				>
-					{ suggestions.map( ( suggestion, index ) => (
-						<LinkControlSearchItem
-							key={ `${ suggestion.id }-${ suggestion.type }` }
-							itemProps={ buildSuggestionItemProps(
-								suggestion,
-								index
-							) }
-							suggestion={ suggestion }
-							onClick={ () => {
-								onChange( { ...value, ...suggestion } );
-								stopEditing();
-							} }
-							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes(
-								suggestion.type.toLowerCase()
-							) }
-							searchTerm={ inputValue }
-						/>
-					) ) }
+					{ suggestions.map( ( suggestion, index ) => {
+						if (
+							shouldShowCreateEntity &&
```

Comment:
I feel like we should never allow the `CREATE_TYPE` suggestion to be included in the suggestions in the first place, if we're not anticipating to be showing the option to create entity. Which has the added benefit of:

1. We don't need this part of the condition, we only need to check if `suggestion.type === CREATE_TYPE`.
2. We can remove the `return null;` condition immediately after this one, because there are only two possible return paths (it is or it isn't `CREATE_TYPE`).

Aside: we should be consistent with our naming, if we're no longer calling it "entity" publicly, then this should be updated as well.

Reply:
> Aside: we should be consistent with our naming, if we're no longer calling it "entity" publicly, then this should be updated as well.

Fixed.

Reply:
Solving this is going to require quite a bit of refactoring. I feel like it's a good thing to do, but as it's not central to this API could/should it be addressed in a follow up PR?